File: src/main/java/com/learnspark/ai/repository/UserAiConfigRepository.java
package com.learnspark.ai.repository;

import com.learnspark.ai.entity.UserAiConfig;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

/**
 * 用户 AI 配置仓库。
 */
@Repository
public interface UserAiConfigRepository extends JpaRepository<UserAiConfig, String> {
}

File: learnspark-kmp/server/src/main/kotlin/com/learnspark/server/domain/repository/UserAiConfigRepository.kt
package com.learnspark.server.domain.repository

import com.learnspark.server.domain.entity.UserAiConfig
import org.springframework.data.jpa.repository.JpaRepository

/**
 * R2：UserAiConfig Repository。
 */
interface UserAiConfigRepository : JpaRepository<UserAiConfig, String> {
    fun findByUserId(userId: String): List<UserAiConfig>
    fun findByUserIdAndProvider(userId: String, provider: String): UserAiConfig?
    fun deleteByUserIdAndProvider(userId: String, provider: String): Long
}
